### PR TITLE
Better API and SDKs documentation

### PR DESCRIPTION
**Pull request by bfirsh**
### Background

I have been working on making the Remote API easier to use. This has involved various things:
- Improving the client libraries we provide for Python and Go, and giving them a name to make them sound like a product we provide – the "Docker SDKs". [The improvements to the Python API are happening here.](https://github.com/docker/docker-py/pull/1186) Go is a WIP.
- [Improving the reference documentation for the Remote API, based on a Swagger definition.](https://bfirsh.github.io/docker-api-reference/)
- Providing an overview to the API and SDKs, and some introductory guides to help users get started. (This PR!)
### What I've done

I've written some better documentation for the Remote API and SDKs. It includes a better high-level overview, and a new getting started guide.

It also links off to the new Swagger-based API reference. I have moved the links to docs for previous API versions to this page so that users immediately land on something useful.

I've discussed some of the docs with @joaofnfernandes who took a look at a very early version. The version in this PR is not complete yet, but I figure it'd be better to open this sooner rather than later so people can start to take a look and be aware of its existence.

I have left in the old documentation for the time being while we figure out what to do with it.
### Todo
- [x] `curl` examples in the getting started guide.
- [x] A call-to-action at the bottom of the getting started guide ("what's next?")
### Dependencies
- [x] [The new Python SDK.](https://github.com/docker/docker-py/pull/1186) (Released in 2.0.0.)
- [x] [Automatically generated API reference.](https://github.com/docker/docker.github.io/pull/606)
### Nice-to-haves
- [ ] Some improvements to the Go SDK, e.g. a `ContainerRun` short-hand. (The Go examples in the getting started guide are a bit of a mouthful.)
### /cc
- @dnephin for the Go SDK and Swagger stuff.
- @shin- for the Python SDK.

### CODE REVIEW

File: api/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+
+To try out the Docker Remote API in development, [you first need to install Docker](https://docs.docker.com/engine/installation/).
+
+Next, you need to install an SDK for the language you are using. There are official ones available for Python and Go, and a number of community maintained libraries for other languages. [Head to the SDKs page to find and install them.](sdks.md)
+
+## Running a container
+
+The most basic thing you can do with Docker is running a container. On the command line, you would use the `docker run` command, but this is just as easy to do from your own apps too.
+
+This is the equivalent of doing `docker run alpine echo hello world`:
```
Comment (joaofnfernandes) [inline api/getting-started.md]: What if we had 4 tabs instead, the default one being CLI? That would probably make it easier to scan with the eyes and find the example you're looking for.

```
 [CLI][Curl][Python][Go]-------------------
 |
 |   docker run alpine echo hello world
 |
 +-----------------------------------------
```

**Comment by johndmulhausen** [comment]
Deploy preview ready!

Built with commit b4449615af39542054d67d7467ac0a2e6779674c

https://deploy-preview-210--docker-docs.netlify.com

**Comment by bfirsh** [comment]
I have now based this on top of https://github.com/docker/docker.github.io/pull/606 and updated it to make sense in context of recent changes. It is essentially an extension of that PR, but with:

 - an introduction page (`engine/api/index.md`)
 - a getting started guide for various languages and curl (`engine/api/getting-started.md`)
 - a much improved client libraries page (`engine/api/sdks.md`).

Once #606 is in, I think this is pretty much ready to go. The only hard dependency is Docker SDK for Python 2.0, which will be released before 1.13 anyway. I recall @johndmulhausen saying we might have to review the design of horizontal tabs, because that hasn't been done before. (In fairness, they do look a bit naff, and are hard to use.)

/cc @johndmulhausen @mstanleyjones

**Comment by mstanleyjones** [comment]
After #606 is in, you'll need to rebase this again before we can merge it. We're getting there!

**Comment by bfirsh** [comment]
@mstanleyjones @johndmulhausen Rebased on top of #606 and `vnext-engine`. Also, the new Docker SDK for Python has been released, so this is all ready to go now. :)

**Comment by mstanleyjones** [comment]
I'm going to merge this, but ping @johndmulhausen to do a follow-up PR to toggle all tabs on a page when one is toggled, and set a cookie so the user's preference is remembered on all API pages. I think the function is `toggle_all_tabs` in `alljs.js`.

**Comment by bfirsh** [comment]
Yay thank you! :tada:

Ideally, clicking a tab would toggle all the ones on the page, but I didn't have time to do it in a non-hacky way, so I just left it out.

**Pull request merged**